scripts/creation-portal-migration/template/generate/gQuestionSet.js

```diff
@@ -0,0 +1,652 @@
+const { ObjectId, ObjectID } = require("mongodb");
+const { createQuestionSet } = require("../../api-list/question");
+
+const { CONFIG } = require("../../constant/config");
+const { findAll, updateById } = require("../../db");
+const logger = require("../../logger");
+const { updateHierarchyTemplate, getPrecondition } = require("../helpers/hierarchyHelper");
+
+const {
+  setQuestionSetTemplate,
+  createQuestionTemplate,
+} = require("../helpers/questionsetHelper");
+const { createProgramTemplate } = require("./gProgram");
+const { getCriteriaData } = require("../migrate/common");
+const { isEmpty, omit, uniq, compact } = require("lodash");
+
+const getQuestionSetTemplates = async (solutions, migratedCount) => {
+  const data = [];
+  // solutions.map(async (solution) => {
+  for (let solution of solutions) {
+    // solution.author = '86d2d978-5b20-4453-8a76-82b5a4c728c9';
+    let programId = solution?.migrationReference?.sourcingProgramId;
```

Comment:
@swetha-aggidevara if solution?.migrationReference?.sourcingProgramId is there do we have to create program ?

Reply:
@VISHNUDAS-tunerlabs  No need to create the program, It is to verify if the program has been created (migrated), update, published, nominated, accepted and contributer added.

In case of any migration failure If the user tries to run the script again, to make sure all the above flow is success it will be called and if any flow is missed it runs that flow.